<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Trajectory inference across conditions: differential expression and differential progression • bioc2020trajectories</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Trajectory inference across conditions: differential expression and differential progression">
<meta property="og:description" content="bioc2020trajectories">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bioc2020trajectories</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.92</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/adventExample.html">Trajectory inference across conditions: toy example</a>
    </li>
    <li>
      <a href="../articles/workshopTrajectories.html">Trajectory inference across conditions: differential expression and differential progression</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kstreet13/bioc2020trajectories">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshopTrajectories_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Trajectory inference across conditions: differential expression and differential progression</h1>
            
      
      
      <div class="hidden name"><code>workshopTrajectories.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<div id="instructors-names-and-contact-information" class="section level2">
<h2 class="hasAnchor">
<a href="#instructors-names-and-contact-information" class="anchor"></a>Instructor(s) name(s) and contact information</h2>
<ul>
<li>Kelly Street (<a href="mailto:kstreet@ds.dfci.harvard.edu" class="email">kstreet@ds.dfci.harvard.edu</a>)</li>
<li>Koen Van den Berge (<a href="mailto:koenvdberge@berkeley.edu" class="email">koenvdberge@berkeley.edu</a>)</li>
<li>Hector Roux de Bézieux (<a href="mailto:hector.rouxdebezieux@berkeley.edu" class="email">hector.rouxdebezieux@berkeley.edu</a>)</li>
</ul>
</div>
<div id="workshop-description" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>In single-cell RNA-sequencing (scRNA-seq), gene expression is assessed at the level of single cells. In dynamic biological systems, it may not be appropriate to assign cells to discrete groups, but rather a continuum of cell states may be observed, e.g. the differentiation of a stem cell population into mature cell types. This is often represented as a trajectory in a reduced dimension of the scRNA-seq dataset.</p>
<p>Many methods have been suggested for trajectory inference. However, in this setting, it is often unclear how one should handle multiple biological groups or conditions, e.g. constructing and comparing the differentiation trajectory of a wild type versus a knock-out stem cell population.</p>
<p>In this workshop, we will explore methods for comparing multiple conditions in a trajectory inference analysis. We start by integrating datasets from multiple conditions into a single trajectory. By comparing the conditions along the trajectory’s path, we can detect large-scale changes, indicative of differential progression. We also demonstrate how to detect subtler changes by finding genes that exhibit different behaviors between these conditions along a differentiation path.</p>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/slingshot/inst/doc/conditionsVignette.html">This vignette</a> provides a more complete problem description and proposes a few analytical approaches, which will serve as the basis of our workshop.</p>
</div>
<div id="pre-requisites" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h2>
<p>Software:</p>
<ul>
<li>Basic knowledge of <em>R</em> syntax</li>
<li>Familiarity with single-cell RNA-sequencing</li>
<li>Familiarity with the <code>SingleCellExperiment</code> class</li>
</ul>
<p>Background reading:</p>
<ul>
<li>The textbook “Orchestrating Single-Cell Analysis with Bioconductor” is a great reference for single-cell analysis using Bioconductor packages.</li>
<li>The data used in this vignette was originally published in <a href="https://www.nature.com/articles/s41588-019-0489-5">McFaline-Figueroa, et al.</a> “A pooled single-cell genetic screen identifies regulatory checkpoints in the continuum of the epithelial-to-mesenchymal transition.”</li>
<li><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4772-0">Slingshot paper</a></li>
<li><a href="https://www.nature.com/articles/s41467-020-14766-3">tradeSeq paper</a></li>
</ul>
</div>
<div id="workshop-participation" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h2>
<p>The workshop will start with an introduction to the problem and the dataset using presentation slides. Following this, we will have a lab session on how one may tackle the problem of handling multiple conditions in trajectory inference and in downstream analysis involving differential progression and differential expression.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level2">
<h2 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h2>
<ul>
<li>The workshop will focus on Bioconductor packages <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html">SingleCellExperiment</a>, <a href="https://bioconductor.org/packages/release/bioc/html/slingshot.html">Slingshot</a>, and <a href="https://bioconductor.org/packages/release/bioc/html/tradeSeq.html">tradeSeq</a>
</li>
</ul>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Data Integration and Trajectory Inference</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Differential Progression</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Differential Expression</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Wrap-up and Conclusions</td>
<td>5m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h2>
<p>Participants will learn how to reason about trajectories in single-cell RNA-seq data and how they may be used for interpretation of complex scRNA-seq datasets.</p>
</div>
<div id="learning-goals" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h2>
<ul>
<li>Reason about dynamic biological systems.</li>
<li>Grasp the complexity of analyzing large scRNA-seq datasets with the goal of extracting relevant biological information.</li>
<li>Understand the concepts of differential progression and differential expression along a trajectory path.</li>
</ul>
</div>
<div id="learning-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h2>
<ul>
<li>Learn how to analyze single-cell RNA-seq data using Bioconductor packages.</li>
<li>Import and explore large scRNA-seq datasets.</li>
<li>Understand the challenges of trajectory inference.</li>
<li>Compose analysis pipeline that allows interpretation of complex scRNA-seq datasets.</li>
<li>Assess the added complexity of handling multiple conditions in these dynamic systems and how it influences the analysis pipeline.</li>
<li>Discuss how the analysis pipeline can incorporate this change and evaluate it.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">slingshot</span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org">scales</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sjmgarnier/viridis">viridis</a></span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://github.com/hms-dbmi/UpSetR">UpSetR</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr">msigdbr</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ctlab/fgsea/">fgsea</a></span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/">knitr</a></span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://statomics.github.io/tradeSeq/index.html">tradeSeq</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis" class="anchor"></a>Analysis</h1>
<div id="dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#dataset" class="anchor"></a>Dataset</h2>
<p>The dataset we will be working with concerns a single-cell RNA-sequencing dataset consisting of two different experiments, which correspond to two treatments. <a href="https://www.nature.com/articles/s41588-019-0489-5">McFaline-Figueroa et al.</a> studied the epithelial-to-mesenchymal transition (EMT), where cells spatially migrate from the epithelium to the mesenchyme during development. This process will be described by a trajectory, reflecting the gene expression changes occurring during this migration. The authors furthermore studied both a control (<code>Mock</code>) condition, and a condition under activation of transforming growth factor <span class="math inline">\(\beta\)</span> (TGFB).</p>
<p>In summary, we will be investigating a trajectory consisting of a single lineage that represents the EMT. This lineage is studied in two different conditions; a control condition and a TGFB-induced condition.</p>
</div>
<div id="integration" class="section level2">
<h2 class="hasAnchor">
<a href="#integration" class="anchor"></a>Integration</h2>
<p>Our dataset contains cells collected from samples undergoing two different treatment conditions which were necessarily collected separately. Hence, we will start with an integration step to combine these two sets of cells, similar to batch correction. Our goal is to remove the technical effects of the different sample collections while preserving any true, biological differences between the two treatment groups.</p>
<p>Data integration and normalization are complex problems and there are a variety of methods addressing each. Interested participants can explore <a href="http://bioconductor.org/books/release/OSCA/integrating-datasets.html">the corresponding chapter of the Bioconductor Ebook</a>. However, since neither is the main focus of this workshop, we elected to use an existing pipeline for these tasks. The full Seurat data integration workflow with SCTransform normalization is described in <a href="https://satijalab.org/seurat/v3.1/integration.html">this vignette</a>.</p>
<p>Since this whole step is quite slow, it will not be run during the workshop but the code is provided below, along with a function to download and preprocess the public data from GEO.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">bioc2020trajectories</span><span class="fu">::</span><span class="fu"><a href="../reference/importRawData.html">importRawData</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">########################</span>
<span class="co">### Split by condition and convert to Seurat</span>
<span class="co">########################</span>
<span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log1p</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>
<span class="va">sceMock</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span> ,<span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">==</span><span class="st">'Mock'</span><span class="op">]</span>
<span class="va">sceTGFB</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span> ,<span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">==</span><span class="st">'TGFB'</span><span class="op">]</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">Seurat</span><span class="op">)</span>
<span class="va">soMock</span> <span class="op">&lt;-</span> <span class="fu">as.Seurat</span><span class="op">(</span><span class="va">sceMock</span><span class="op">)</span>
<span class="va">soTGFB</span> <span class="op">&lt;-</span> <span class="fu">as.Seurat</span><span class="op">(</span><span class="va">sceTGFB</span><span class="op">)</span>

<span class="co">########################</span>
<span class="co">### Normalize</span>
<span class="co">########################</span>
<span class="va">soMock</span> <span class="op">&lt;-</span> <span class="fu">SCTransform</span><span class="op">(</span><span class="va">soMock</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">soTGFB</span> <span class="op">&lt;-</span> <span class="fu">SCTransform</span><span class="op">(</span><span class="va">soTGFB</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co">########################</span>
<span class="co">### Integrate</span>
<span class="co">########################</span>
<span class="va">dtlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mock <span class="op">=</span> <span class="va">soMock</span>, TGFB <span class="op">=</span> <span class="va">soTGFB</span><span class="op">)</span>
<span class="va">intfts</span> <span class="op">&lt;-</span> <span class="fu">SelectIntegrationFeatures</span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">dtlist</span>, nfeatures <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span> <span class="co"># maxes out at 4080 (why?)</span>
<span class="va">dtlist</span> <span class="op">&lt;-</span> <span class="fu">PrepSCTIntegration</span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">dtlist</span>,
                             anchor.features <span class="op">=</span> <span class="va">intfts</span><span class="op">)</span>
<span class="va">anchors</span> <span class="op">&lt;-</span> <span class="fu">FindIntegrationAnchors</span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">dtlist</span>, normalization.method <span class="op">=</span> <span class="st">"SCT"</span>,
                                  anchor.features <span class="op">=</span> <span class="va">intfts</span><span class="op">)</span>
<span class="va">integrated</span> <span class="op">&lt;-</span> <span class="fu">IntegrateData</span><span class="op">(</span>anchorset <span class="op">=</span> <span class="va">anchors</span>, normalization.method <span class="op">=</span> <span class="st">"SCT"</span><span class="op">)</span>
<span class="va">integrated</span> <span class="op">&lt;-</span> <span class="fu">RunPCA</span><span class="op">(</span><span class="va">integrated</span><span class="op">)</span>
<span class="va">integrated</span> <span class="op">&lt;-</span> <span class="fu">RunUMAP</span><span class="op">(</span><span class="va">integrated</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span>

<span class="co">## convert back to singleCellExperiment</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">as.SingleCellExperiment</span><span class="op">(</span><span class="va">integrated</span>, assay<span class="op">=</span><span class="st">"RNA"</span><span class="op">)</span></code></pre></div>
</div>
<div id="import-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#import-dataset" class="anchor"></a>Import dataset</h2>
<p>We have made the pre-processed, integrated dataset available as a <code>SingleCellExperiment</code> object in the workshop package, which we import below.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"sce"</span>, package <span class="op">=</span> <span class="st">"bioc2020trajectories"</span><span class="op">)</span></code></pre></div>
</div>
<div id="differential-topology" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-topology" class="anchor"></a>Differential Topology</h2>
<p>Once the two datasets have been integrated, we can visualize all the single cells in a shared reduced dimensional space. We also visualize the distribution of cells in this space according to the treatment (control and TGFB) and spatial location (inner cells versus outer cells).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">shuffle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/layout.html">layout</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4.5</span>,<span class="fl">4</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDims</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">UMAP</span><span class="op">[</span><span class="va">shuffle</span>, <span class="op">]</span>,
  asp <span class="op">=</span> <span class="fl">1</span>, pch <span class="op">=</span> <span class="fl">16</span>, xlab <span class="op">=</span> <span class="st">"UMAP-1"</span>, ylab <span class="op">=</span> <span class="st">"UMAP-2"</span>,
  col <span class="op">=</span> <span class="fu"><a href="https://scales.r-lib.org//reference/alpha.html">alpha</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="va">shuffle</span><span class="op">]</span>, alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, pch <span class="op">=</span> <span class="fl">16</span>, col <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, bty <span class="op">=</span> <span class="st">"n"</span>, 
       legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDims</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">UMAP</span><span class="op">[</span><span class="va">shuffle</span>, <span class="op">]</span>, asp <span class="op">=</span> <span class="fl">1</span>, pch <span class="op">=</span> <span class="fl">16</span>, xlab <span class="op">=</span> <span class="st">"UMAP-1"</span>, ylab <span class="op">=</span> <span class="st">"UMAP-2"</span>, 
     col <span class="op">=</span> <span class="fu"><a href="https://scales.r-lib.org//reference/alpha.html">alpha</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">spatial_id</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="va">shuffle</span><span class="op">]</span>, alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, pch <span class="op">=</span> <span class="fl">16</span>, col <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">4</span>, bty <span class="op">=</span> <span class="st">"n"</span>, legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">spatial_id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/compute%20scores-1.png" width="672"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/layout.html">layout</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fl">.1</span><span class="op">)</span></code></pre></div>
<p>We know from biological knowledge that the EMT development goes from the inner to the outer cells. The question is: should we fit a separate trajectory for each condition? We might expect the trajectory itself to be changed by the treatment if the treatment effect is systematically large. Otherwise, the treatment may impact the expression profile of some genes but the overall trajectory will be preserved.</p>
<p>To help assess this, we devised an imbalance score. Regions with a high score indicate that the local cell distribution according to treatment label is unbalanced compared the overall distribution. Here, we see that, while there are some small regions of imbalance, the global path along the development axis is well-balanced. This means that we can fit a global trajectory to the full dataset. This choice allows us to use the maximal amount of data in the construction of our trajectory, which should lead to more robust results than separate, potentially noisy trajectories constructed on subsets of the data. As we will see, not all cell types are necessarily present in every condition, so this approach ensures that our trajectory accounts for all cell types present in the overall data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu">bioc2020trajectories</span><span class="fu">::</span><span class="fu"><a href="../reference/imbalance_score.html">imbalance_score</a></span><span class="op">(</span>
  rd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDims</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">UMAP</span>, 
  cl <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span>,
  k <span class="op">=</span> <span class="fl">20</span>, smooth <span class="op">=</span> <span class="fl">40</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: replacing previous import 'HDF5Array::path' by 'igraph::path' when
## loading 'bioc2020trajectories'</code></pre>
<pre><code>## Warning: replacing previous import 'igraph::compose' by 'purrr::compose' when
## loading 'bioc2020trajectories'</code></pre>
<pre><code>## Warning: replacing previous import 'igraph::simplify' by 'purrr::simplify' when
## loading 'bioc2020trajectories'</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">grad</span> <span class="op">&lt;-</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">plasma</a></span><span class="op">(</span><span class="fl">10</span>, begin <span class="op">=</span> <span class="fl">0</span>, end <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">grad</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cut.html">cut</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">scaled_scores</span>, breaks <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDims</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">UMAP</span>, col <span class="op">=</span> <span class="va">grad</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/cut.html">cut</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">scaled_scores</span>, breaks <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">]</span>,
     asp <span class="op">=</span> <span class="fl">1</span>, pch <span class="op">=</span> <span class="fl">16</span>, xlab <span class="op">=</span> <span class="st">"UMAP-1"</span>, ylab <span class="op">=</span> <span class="st">"UMAP-2"</span>, cex <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span><span class="op">(</span><span class="st">"topleft"</span>, legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">grad</span><span class="op">)</span>, col <span class="op">=</span> <span class="va">grad</span>, pch <span class="op">=</span> <span class="fl">16</span>, bty <span class="op">=</span> <span class="st">"n"</span>, cex <span class="op">=</span> <span class="fl">2</span> <span class="op">/</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p>For more information on the score, run <code><a href="../reference/imbalance_score.html">help("imbalance_score", "bioc2020trajectories")</a></code></p>
</div>
<div id="trajectory-inference" class="section level2">
<h2 class="hasAnchor">
<a href="#trajectory-inference" class="anchor"></a>Trajectory Inference</h2>
<p>We perform trajectory inference to order the cells according to EMT progression. We use <code>slingshot</code> for trajectory inference, with the cells’ position (inner or outer) serving as the cluster identifier. This ensures that we will only find a single lineage while still allowing sufficient flexibility to correctly orient the pseudotime axis.</p>
<p>Note that we perform trajectory inference using cells from both conditions, rather than splitting the data into two groups, as discussed above.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">slingshot</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingshot.html">slingshot</a></span><span class="op">(</span><span class="va">sce</span>, reducedDim <span class="op">=</span> <span class="st">'UMAP'</span>, clusterLabels <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">spatial_id</span>,
                 start.clus <span class="op">=</span> <span class="st">'inner'</span>, approx_points <span class="op">=</span> <span class="fl">150</span><span class="op">)</span></code></pre></div>
<pre><code>## Using full covariance matrix</code></pre>
<p><img src="workshopTrajectories_files/figure-html/plotSlingshot-1.png" width="768"></p>
</div>
<div id="differential-progression" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-progression" class="anchor"></a>Differential progression</h2>
<p>Now that we have ordered the cells by EMT progression, we can begin to address the main question: how is this progression affected by TGF-<span class="math inline">\(\beta\)</span> treatment? In this section, we interpret this question as a univariate analysis of the pseudotime values between the two groups.</p>
<p><img src="workshopTrajectories_files/figure-html/plotDensities-1.png" width="700"></p>
<p>The density estimates for the two groups show a trimodal distribution for the untreated cells, but a tendency toward later pseudotime values in the TGF-<span class="math inline">\(\beta\)</span> treated cells. The difference is striking enough that a standard T-test would likely be significant, but we are we are interested more generally in differences between the two distributions, not just the difference of means (one could imagine a scenario in which the treated group tended toward the extremes, but the means were the same). Hence, we propose a Kolmogorov-Smirnov Test to assess whether the two groups of pseudotime values are derived from the same distribution. For more info on the Kolmogorov-Smirnov Test, see <a href="https://en.wikipedia.org/wiki/Kolmogorov%E2%80%93Smirnov_test">here</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">########################</span>
<span class="co">### Kolmogorov-Smirnov Test</span>
<span class="co">########################</span>
<span class="fu"><a href="https://rdrr.io/r/stats/ks.test.html">ks.test</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html">slingPseudotime</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span> <span class="op">==</span> <span class="st">"Mock"</span>, <span class="fl">1</span><span class="op">]</span>,
        <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html">slingPseudotime</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span> <span class="op">==</span> <span class="st">"TGFB"</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p>As we might expect from the plot, this test is highly significant, so we can conclude that there are differences between the two distributions.</p>
</div>
<div id="differential-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential expression</h2>
<p>We will now proceed to discover genes whose expression is associated with the inferred trajectory. We will look for genes that (i) change in gene expression along the trajectory, and (ii) are differentially expressed between the two conditions along the trajectory. The differential expression analysis uses the Bioconductor package <code>tradeSeq</code>. This analysis relies on a new version of <code>tradeSeq</code>, which we have recently updated to allow for multiple conditions.</p>
<p>For each condition (i.e., control and TGF-Beta), a smooth average expression profile along pseudotime will be estimated for each gene, using a negative binomial generalized additive model (NB-GAM). Each differential expression hypothesis of interest will then be translated into testing specific features (a linear combination of the parameters) of this smoothed expression estimate. See <a href="https://www.maths.ed.ac.uk/~swood34/mgcv/">here</a> for more information on smoothers and the <code>mgcv</code> package which we are using to estimate the smoothers.</p>
<p>The next two paragraphs can be time-consuming so we will not run them during the workshop, however, their output is already present in the data object that was loaded at the start of this workshop. They can be easily parallelized, relying on the <code>BiocParallel</code> bioconductor package. See <a href="https://statomics.github.io/tradeSeq/articles/fitGAM.html#parallel-computing-1">here</a> for more details.</p>
<div id="select-number-of-knots" class="section level3">
<h3 class="hasAnchor">
<a href="#select-number-of-knots" class="anchor"></a>Select number of knots</h3>
<p>Before we can fit these smoothed expression profiles, we need to get a sense of how complex the expression patterns are in this dataset. This is translated into selecting a number of knots for the NB-GAMs, where a higher number of knots allows for more complex expression patterns. Here, we pick <span class="math inline">\(5\)</span> knots.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>
<span class="va">icMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/evaluateK.html">evaluateK</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,
                   pseudotime <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">pseudotime</span>,
                   cellWeights <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">cellWeights.V1</span>,
                   conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">)</span>,
                   nGenes <span class="op">=</span> <span class="fl">300</span>,
                   k <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></code></pre></div>
<p><img src="../data/evaluateKResult.png"><!-- --></p>
<p>The plot above shows the graphical output from running <code>evaluateK</code>. The left panel shows the distribution of gene-level AIC values as compared to their average AIC over the range of <code>k</code>. The second and third panel plot the average AIC and relative AIC with respect to the lowest value of <code>k</code> (i.e., 3), respectively, as a function of <code>k</code>. Finally, the right panel plots the number of genes whose AIC is lowest at a particular value of <code>k</code>.</p>
<p>Choosing an appropriate value of <code>k</code> can be seen as analogous to choosing the number of principal components based on a scree plot: we look for an ‘elbow point’, where the decrease starts attenuating. Here, we choose <code>k=5</code> to allow for flexible, yet simple, functions while limiting the computational burden. In general, we found the influence of choosing the exact value of <code>k</code> to be rather limited, unless <code>k</code> is arbitrarily small or large. In our evaluations, most datasets fall within the range of <span class="math inline">\(4\)</span> to <span class="math inline">\(8\)</span> knots.</p>
</div>
<div id="fit-gam" class="section level3">
<h3 class="hasAnchor">
<a href="#fit-gam" class="anchor"></a>Fit GAM</h3>
<p>Next, we fit the NB-GAMs using 5 knots, based on the pseudotime and cell-level weights estimated by Slingshot. We use the <code>conditions</code> argument to fit separate smoothers for each condition.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/fitGAM.html">fitGAM</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,
              pseudotime <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">pseudotime</span>,
              cellWeights <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">cellWeights.V1</span>,
              conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">)</span>,
              nknots <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">tradeSeq</span><span class="op">$</span><span class="va">converged</span><span class="op">)</span></code></pre></div>
</div>
<div id="assess-de-along-pseudotime-or-pseudospace" class="section level3">
<h3 class="hasAnchor">
<a href="#assess-de-along-pseudotime-or-pseudospace" class="anchor"></a>Assess DE along pseudotime (or pseudospace)</h3>
<p>Note that the axis represented by the trajectory in this dataset is actually the migration of cells from the epithelium to the mesenchyme and therefore could also be looked at as a space dimension, although it is likely to be correlated with chronological time, too.</p>
<p>To assess significant changes in gene expression as a function of pseudotime within each lineage, we use the <code>associationTest</code>, which tests the null hypothesis that gene expression is not a function of pseudotime, i.e., whether the estimated smoothers are significantly varying as a function of pseudotime within each lineage. The <code>lineages=TRUE</code> argument specifies that we would like the results for each lineage separately, asides from the default global test, which tests for significant associations across all lineages in the trajectory simultaneously. Further, we specify a log2 fold change cut-off to test against using the <code>l2fc</code> argument.</p>
<p>On a lineage-specific basis, there are over twice as much DE genes in the mock lineage (2398) as compared to the TGFB lineage (1013). Many of the DE genes in the TGFB condition are also DE in the mock condition, around 80%.</p>
<p>The authors of the original paper found <span class="math inline">\(1105\)</span> DE genes for the mock condition on a FDR level of <span class="math inline">\(1e-10\)</span> and a cut-off of 1 on the absolute value of the log2 fold change.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">assocRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/associationTest.html">associationTest</a></span><span class="op">(</span><span class="va">sce</span>, lineages <span class="op">=</span> <span class="cn">TRUE</span>, l2fc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://statomics.github.io/tradeSeq/index.html">tradeSeq</a></span><span class="op">)</span>
<span class="va">assocRes</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">assocRes</span>
<span class="va">mockGenes</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">assocRes</span><span class="op">)</span><span class="op">[</span>
  <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">assocRes</span><span class="op">$</span><span class="va">pvalue_lineage1_conditionMock</span>, <span class="st">"fdr"</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="op">]</span>
<span class="va">tgfbGenes</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">assocRes</span><span class="op">)</span><span class="op">[</span>
  <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">assocRes</span><span class="op">$</span><span class="va">pvalue_lineage1_conditionTGFB</span>, <span class="st">"fdr"</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">mockGenes</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 2398</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">tgfbGenes</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1013</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">UpSetR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/UpSetR/man/upset.html">upset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/UpSetR/man/fromList.html">fromList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>mock <span class="op">=</span> <span class="va">mockGenes</span>, tgfb <span class="op">=</span> <span class="va">tgfbGenes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div id="visualization-of-de-genes" class="section level4">
<h4 class="hasAnchor">
<a href="#visualization-of-de-genes" class="anchor"></a>Visualization of DE genes</h4>
<p>Below we visualize and cluster the genes whose expression vary over pseudotime, using the smoothed expression patterns As was also observed in the original manuscript, genes are mainly upregulated at the start- or endpoints of the lineage.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### based on mean smoother</span>
<span class="va">yhatSmooth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/predictSmooth.html">predictSmooth</a></span><span class="op">(</span><span class="va">sce</span>, gene <span class="op">=</span> <span class="va">mockGenes</span>, nPoints <span class="op">=</span> <span class="fl">50</span>, tidy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">heatSmooth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">yhatSmooth</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
                       cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
                       show_rownames <span class="op">=</span> <span class="cn">FALSE</span>, 
                       show_colnames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## the hierarchical trees constructed here, can also be used for </span>
<span class="co">## clustering of the genes according to their average expression pattern.</span>
<span class="va">cl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span><span class="op">(</span><span class="va">heatSmooth</span><span class="op">$</span><span class="va">tree_row</span>, k <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cl</span><span class="op">)</span></code></pre></div>
<pre><code>## cl
##   1   2   3   4   5   6 
## 269 778 197 740 338  76</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conditions</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span>
<span class="va">pt1</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">pseudotime</span>
  
<span class="co">### based on fitted values (plotting takes a while to run)</span>
<span class="va">yhatCell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/predictCells.html">predictCells</a></span><span class="op">(</span><span class="va">sce</span>, gene<span class="op">=</span><span class="va">mockGenes</span><span class="op">)</span>
<span class="va">yhatCellMock</span> <span class="op">&lt;-</span> <span class="va">yhatCell</span><span class="op">[</span>,<span class="va">conditions</span> <span class="op">==</span> <span class="st">"Mock"</span><span class="op">]</span>
<span class="co"># order according to pseudotime</span>
<span class="va">ooMock</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pt1</span><span class="op">[</span><span class="va">conditions</span> <span class="op">==</span> <span class="st">"Mock"</span><span class="op">]</span>, decreasing<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">yhatCellMock</span> <span class="op">&lt;-</span> <span class="va">yhatCellMock</span><span class="op">[</span>,<span class="va">ooMock</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">yhatCellMock</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
          show_rownames <span class="op">=</span> <span class="cn">FALSE</span>, show_colnames<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
<div id="gene-set-enrichment-analysis-on-genes-from-the-mock-condition" class="section level4">
<h4 class="hasAnchor">
<a href="#gene-set-enrichment-analysis-on-genes-from-the-mock-condition" class="anchor"></a>Gene set enrichment analysis on genes from the Mock condition</h4>
<p>Gene set enrichment analysis on the DE genes within the mock condition confirms the biology on epithelial cell differentiation.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## C5 category is according to gene ontology grouping: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4707969/pdf/nihms-743907.pdf</span>
<span class="va">geneSets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span>, category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span>
<span class="co">### filter background to only include genes that we assessed.</span>
<span class="va">geneSets</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="va">geneSets</span><span class="op">$</span><span class="va">gene_symbol</span><span class="op">)</span>
<span class="va">geneSets</span> <span class="op">&lt;-</span> <span class="va">geneSets</span><span class="op">[</span><span class="va">geneSets</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">m_list</span> <span class="op">&lt;-</span> <span class="va">geneSets</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span>
<span class="va">stats</span> <span class="op">&lt;-</span> <span class="va">assocRes</span><span class="op">$</span><span class="va">waldStat_lineage1_conditionMock</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">assocRes</span><span class="op">)</span>
<span class="va">eaRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html">fgsea</a></span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">m_list</span>, stats <span class="op">=</span> <span class="va">stats</span>, nperm <span class="op">=</span> <span class="fl">5e4</span>, minSize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in fgsea(pathways = m_list, stats = stats, nperm = 50000, minSize = 10):
## You are trying to run fgseaSimple. It is recommended to use fgseaMultilevel. To
## run fgseaMultilevel, you need to remove the nperm argument in the fgsea function
## call.</code></pre>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.02% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ooEA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">eaRes</span><span class="op">$</span><span class="va">pval</span>, decreasing <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">eaRes</span><span class="op">[</span><span class="va">ooEA</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="74%">
<col width="12%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left">pathway</th>
<th align="right">pval</th>
<th align="right">padj</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GO_CELL_CYCLE</td>
<td align="right">0.0000600</td>
<td align="right">0.1029839</td>
</tr>
<tr class="even">
<td align="left">GO_EPIDERMIS_DEVELOPMENT</td>
<td align="right">0.0000800</td>
<td align="right">0.1029839</td>
</tr>
<tr class="odd">
<td align="left">GO_MITOTIC_CELL_CYCLE</td>
<td align="right">0.0000800</td>
<td align="right">0.1029839</td>
</tr>
<tr class="even">
<td align="left">GO_SKIN_DEVELOPMENT</td>
<td align="right">0.0001200</td>
<td align="right">0.1029839</td>
</tr>
<tr class="odd">
<td align="left">GO_EPITHELIUM_DEVELOPMENT</td>
<td align="right">0.0001400</td>
<td align="right">0.1029839</td>
</tr>
<tr class="even">
<td align="left">GO_EPITHELIAL_CELL_DIFFERENTIATION</td>
<td align="right">0.0001800</td>
<td align="right">0.1029839</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_MODIFICATION_OF_SYNAPTIC_STRUCTURE</td>
<td align="right">0.0002003</td>
<td align="right">0.1029839</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_CHROMOSOME_ORGANIZATION</td>
<td align="right">0.0002200</td>
<td align="right">0.1029839</td>
</tr>
<tr class="odd">
<td align="left">GO_MITOTIC_NUCLEAR_DIVISION</td>
<td align="right">0.0002600</td>
<td align="right">0.1029839</td>
</tr>
<tr class="even">
<td align="left">GO_NEGATIVE_REGULATION_OF_CELLULAR_COMPONENT_ORGANIZATION</td>
<td align="right">0.0002600</td>
<td align="right">0.1029839</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEOLYSIS</td>
<td align="right">0.0003000</td>
<td align="right">0.1056246</td>
</tr>
<tr class="even">
<td align="left">GO_ORGANELLE_FISSION</td>
<td align="right">0.0003200</td>
<td align="right">0.1056246</td>
</tr>
<tr class="odd">
<td align="left">GO_EPIDERMAL_CELL_DIFFERENTIATION</td>
<td align="right">0.0004400</td>
<td align="right">0.1340619</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_CYCLE_PROCESS</td>
<td align="right">0.0005200</td>
<td align="right">0.1452338</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_NUCLEAR_DIVISION</td>
<td align="right">0.0005600</td>
<td align="right">0.1452338</td>
</tr>
<tr class="even">
<td align="left">GO_NEGATIVE_REGULATION_OF_ORGANELLE_ORGANIZATION</td>
<td align="right">0.0006000</td>
<td align="right">0.1452338</td>
</tr>
<tr class="odd">
<td align="left">GO_KERATINOCYTE_DIFFERENTIATION</td>
<td align="right">0.0006400</td>
<td align="right">0.1452338</td>
</tr>
<tr class="even">
<td align="left">GO_CHROMOSOME_ORGANIZATION</td>
<td align="right">0.0006600</td>
<td align="right">0.1452338</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_ORGANELLE_ORGANIZATION</td>
<td align="right">0.0007600</td>
<td align="right">0.1584368</td>
</tr>
<tr class="even">
<td align="left">GO_CHROMOSOME_SEGREGATION</td>
<td align="right">0.0009000</td>
<td align="right">0.1782414</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="differential-expression-between-conditions" class="section level3">
<h3 class="hasAnchor">
<a href="#differential-expression-between-conditions" class="anchor"></a>Differential expression between conditions</h3>
<p>In the following sections, we will investigate differential expression for each gene, between the two conditions.</p>
<p>We will first make exploratory data analysis visualizations to take a look at the expression patterns of genes that were also discussed in the original manuscript. The paper mentions that CDH1 and CRB3 should be expressed in similar kinetics. Note that the lower slope of CDH1 is also observed in the paper.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">sce</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>, gene <span class="op">=</span> <span class="st">"CDH1"</span>, alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"CDH1"</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">sce</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>, gene <span class="op">=</span> <span class="st">"CRB3"</span>, alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"CRB3"</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<p>They also mention that ‘only cells treated with TGF-Beta and positioned at the outer extreme of the trajectory expressed robust levels of FN1 and CDH2’.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">sce</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>, gene <span class="op">=</span> <span class="st">"FN1"</span>, alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"FN1"</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">sce</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>, gene <span class="op">=</span> <span class="st">"CDH2"</span>, alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"CDH2"</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-12-2.png" width="700"></p>
<div id="differential-expression-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential expression analysis</h4>
<p>To test differential expression between conditions, we use the <code>conditionTest</code> function implemented in <code>tradeSeq</code>. This function tests the null hypothesis that genes have identical expression patterns in each condition. We discover <span class="math inline">\(1993\)</span> genes that are DE with a fold change higher than <span class="math inline">\(2\)</span> or lower than <span class="math inline">\(1/2\)</span>.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">condRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/conditionTest.html">conditionTest</a></span><span class="op">(</span><span class="va">sce</span>, l2fc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">pvalue</span>, <span class="st">"fdr"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.1890175</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1993</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conditionGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">)</span><span class="op">[</span><span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">]</span>
<span class="va">conditionGenes</span> <span class="op">&lt;-</span> <span class="va">conditionGenes</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">conditionGenes</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div id="visualize-most-and-least-significant-gene" class="section level4">
<h4 class="hasAnchor">
<a href="#visualize-most-and-least-significant-gene" class="anchor"></a>Visualize most and least significant gene</h4>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot genes</span>
<span class="va">oo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">waldStat</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># most significant gene</span>
<span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">sce</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>,
              gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
              alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># least significant gene</span>
<span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">sce</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>,
              gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">]</span><span class="op">]</span>,
              alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
</div>
</div>
<div id="heatmaps-of-genes-de-between-conditions" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmaps-of-genes-de-between-conditions" class="anchor"></a>Heatmaps of genes DE between conditions</h3>
<p>Below we show heatmaps of the genes DE between conditions. The DE genes in the heatmaps are ordered according to a hierarchical clustering on the TGF-Beta condition.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### based on mean smoother</span>
<span class="va">yhatSmooth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/predictSmooth.html">predictSmooth</a></span><span class="op">(</span><span class="va">sce</span>, gene <span class="op">=</span> <span class="va">conditionGenes</span>, nPoints <span class="op">=</span> <span class="fl">50</span>, tidy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">yhatSmoothScaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">yhatSmooth</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">heatSmooth_TGF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="va">yhatSmoothScaled</span><span class="op">[</span>, <span class="fl">51</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span>,
  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
  show_rownames <span class="op">=</span> <span class="cn">FALSE</span>, show_colnames <span class="op">=</span> <span class="cn">FALSE</span>, main <span class="op">=</span> <span class="st">"TGF-Beta"</span>, legend <span class="op">=</span> <span class="cn">FALSE</span>,
  silent <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="va">matchingHeatmap_mock</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="va">yhatSmoothScaled</span><span class="op">[</span><span class="va">heatSmooth_TGF</span><span class="op">$</span><span class="va">tree_row</span><span class="op">$</span><span class="va">order</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>,
  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>, cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,
  show_rownames <span class="op">=</span> <span class="cn">FALSE</span>, show_colnames <span class="op">=</span> <span class="cn">FALSE</span>, main <span class="op">=</span> <span class="st">"Mock"</span>,
  legend <span class="op">=</span> <span class="cn">FALSE</span>, silent <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">heatSmooth_TGF</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span>, <span class="va">matchingHeatmap_mock</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="workshopTrajectories_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div id="gene-set-enrichment-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-set-enrichment-analysis" class="anchor"></a>Gene set enrichment analysis</h3>
<p>Gene set enrichment analysis on genes that are differentially expressed between conditions finds evidence for cell motility, cell junctions/adhesion and gastrulation. The original paper also focuses on the KRAS signaling pathway, which induces cell migration, amongst others. Other related processes include morphogenesis, gastrulation and cell adhesion.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">statsCond</span> <span class="op">&lt;-</span> <span class="va">condRes</span><span class="op">$</span><span class="va">waldStat</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">statsCond</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">)</span>
<span class="va">eaRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html">fgsea</a></span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">m_list</span>, stats <span class="op">=</span> <span class="va">statsCond</span>, nperm <span class="op">=</span> <span class="fl">5e4</span>, minSize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in fgsea(pathways = m_list, stats = statsCond, nperm = 50000,
## minSize = 10): You are trying to run fgseaSimple. It is recommended to use
## fgseaMultilevel. To run fgseaMultilevel, you need to remove the nperm argument
## in the fgsea function call.</code></pre>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.04% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ooEA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">eaRes</span><span class="op">$</span><span class="va">pval</span>, decreasing <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">eaRes</span><span class="op">[</span><span class="va">ooEA</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="75%">
<col width="12%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left">pathway</th>
<th align="right">pval</th>
<th align="right">padj</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GO_BIOLOGICAL_ADHESION</td>
<td align="right">0.0000200</td>
<td align="right">0.0528523</td>
</tr>
<tr class="even">
<td align="left">GO_ANATOMICAL_STRUCTURE_FORMATION_INVOLVED_IN_MORPHOGENESIS</td>
<td align="right">0.0000400</td>
<td align="right">0.0528523</td>
</tr>
<tr class="odd">
<td align="left">GO_CELL_MOTILITY</td>
<td align="right">0.0000400</td>
<td align="right">0.0528523</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_CELL_ADHESION</td>
<td align="right">0.0001600</td>
<td align="right">0.1585568</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_MITOCHONDRIAL_GENE_EXPRESSION</td>
<td align="right">0.0002600</td>
<td align="right">0.2061239</td>
</tr>
<tr class="even">
<td align="left">GO_TRANSCRIPTION_BY_RNA_POLYMERASE_III</td>
<td align="right">0.0004200</td>
<td align="right">0.2774745</td>
</tr>
<tr class="odd">
<td align="left">GO_EXTRACELLULAR_STRUCTURE_ORGANIZATION</td>
<td align="right">0.0006400</td>
<td align="right">0.3212862</td>
</tr>
<tr class="even">
<td align="left">GO_MITOCHONDRIAL_RNA_METABOLIC_PROCESS</td>
<td align="right">0.0008800</td>
<td align="right">0.3212862</td>
</tr>
<tr class="odd">
<td align="left">GO_VASCULATURE_DEVELOPMENT</td>
<td align="right">0.0009400</td>
<td align="right">0.3212862</td>
</tr>
<tr class="even">
<td align="left">GO_CIRCULATORY_SYSTEM_DEVELOPMENT</td>
<td align="right">0.0009600</td>
<td align="right">0.3212862</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_CELL_POPULATION_PROLIFERATION</td>
<td align="right">0.0010800</td>
<td align="right">0.3212862</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_MORPHOGENESIS_INVOLVED_IN_DIFFERENTIATION</td>
<td align="right">0.0011000</td>
<td align="right">0.3212862</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_CELLULAR_COMPONENT_MOVEMENT</td>
<td align="right">0.0011000</td>
<td align="right">0.3212862</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_MORPHOGENESIS</td>
<td align="right">0.0011800</td>
<td align="right">0.3212862</td>
</tr>
<tr class="odd">
<td align="left">GO_TAXIS</td>
<td align="right">0.0013000</td>
<td align="right">0.3212862</td>
</tr>
<tr class="even">
<td align="left">GO_N_TERMINAL_PROTEIN_AMINO_ACID_MODIFICATION</td>
<td align="right">0.0013600</td>
<td align="right">0.3212862</td>
</tr>
<tr class="odd">
<td align="left">GO_CELL_CELL_ADHESION</td>
<td align="right">0.0014600</td>
<td align="right">0.3212862</td>
</tr>
<tr class="even">
<td align="left">GO_MOTOR_NEURON_AXON_GUIDANCE</td>
<td align="right">0.0014601</td>
<td align="right">0.3212862</td>
</tr>
<tr class="odd">
<td align="left">GO_TUBE_DEVELOPMENT</td>
<td align="right">0.0015400</td>
<td align="right">0.3212862</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_JUNCTION_ASSEMBLY</td>
<td align="right">0.0017600</td>
<td align="right">0.3322143</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="final-notes" class="section level1">
<h1 class="hasAnchor">
<a href="#final-notes" class="anchor"></a>Final notes</h1>
<p>A compiled version of the vignette is available on the <a href="https://kstreet13.github.io/bioc2020trajectories/articles/workshopTrajectories.html">workshop website</a>.</p>
<p>If you have questions that you could not ask during the workshop, feel free to open an issue on the github repository <a href="https://github.com/kstreet13/bioc2020trajectories/issues">here</a>.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R Under development (unstable) (2020-11-02 r79396)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] tradeSeq_1.5.0              gridExtra_2.3              
##  [3] ggplot2_3.3.2               knitr_1.30                 
##  [5] fgsea_1.17.0                msigdbr_7.2.1              
##  [7] pheatmap_1.0.12             UpSetR_1.4.0               
##  [9] viridis_0.5.1               viridisLite_0.3.0          
## [11] scales_1.1.1                RColorBrewer_1.1-2         
## [13] SingleCellExperiment_1.13.3 SummarizedExperiment_1.21.1
## [15] Biobase_2.51.0              GenomicRanges_1.43.1       
## [17] GenomeInfoDb_1.27.3         IRanges_2.25.5             
## [19] S4Vectors_0.29.6            BiocGenerics_0.37.0        
## [21] MatrixGenerics_1.3.0        matrixStats_0.57.0         
## [23] slingshot_1.9.1             princurve_2.1.5            
## 
## loaded via a namespace (and not attached):
##  [1] Rtsne_0.15                    VGAM_1.1-4                   
##  [3] colorspace_2.0-0              ellipsis_0.3.1               
##  [5] rprojroot_2.0.2               XVector_0.31.1               
##  [7] fs_1.5.0                      farver_2.0.3                 
##  [9] ggrepel_0.9.0                 bit64_4.0.5                  
## [11] splines_4.1.0                 docopt_0.7.1                 
## [13] cluster_2.1.0                 dbplyr_2.0.0                 
## [15] bioc2020trajectories_0.0.0.92 HDF5Array_1.19.0             
## [17] compiler_4.1.0                httr_1.4.2                   
## [19] assertthat_0.2.1              Matrix_1.2-18                
## [21] limma_3.47.1                  htmltools_0.5.0              
## [23] tools_4.1.0                   igraph_1.2.6                 
## [25] gtable_0.3.0                  glue_1.4.2                   
## [27] GenomeInfoDbData_1.2.4        RANN_2.6.1                   
## [29] reshape2_1.4.4                dplyr_1.0.2                  
## [31] rappdirs_0.3.1                fastmatch_1.1-0              
## [33] Rcpp_1.0.5                    slam_0.1-48                  
## [35] DDRTree_0.1.5                 pkgdown_1.6.1                
## [37] rhdf5filters_1.3.3            vctrs_0.3.5                  
## [39] ape_5.4-1                     nlme_3.1-151                 
## [41] xfun_0.19                     stringr_1.4.0                
## [43] lifecycle_0.2.0               irlba_2.3.3                  
## [45] edgeR_3.33.0                  zlibbioc_1.37.0              
## [47] ragg_0.4.0                    rhdf5_2.35.0                 
## [49] monocle_2.19.0                curl_4.3                     
## [51] yaml_2.2.1                    memoise_1.1.0                
## [53] pbapply_1.4-3                 fastICA_1.2-2                
## [55] stringi_1.5.3                 RSQLite_2.2.1                
## [57] highr_0.8                     desc_1.2.0                   
## [59] filelock_1.0.2                densityClust_0.3             
## [61] BiocParallel_1.25.1           rlang_0.4.9                  
## [63] pkgconfig_2.0.3               systemfonts_0.3.2            
## [65] bitops_1.0-6                  qlcMatrix_0.9.7              
## [67] evaluate_0.14                 lattice_0.20-41              
## [69] Rhdf5lib_1.13.0               purrr_0.3.4                  
## [71] labeling_0.4.2                bit_4.0.4                    
## [73] tidyselect_1.1.0              plyr_1.8.6                   
## [75] magrittr_2.0.1                R6_2.5.0                     
## [77] generics_0.1.0                combinat_0.0-8               
## [79] DelayedArray_0.17.6           DBI_1.1.0                    
## [81] pillar_1.4.7                  withr_2.3.0                  
## [83] mgcv_1.8-33                   RCurl_1.98-1.2               
## [85] tibble_3.0.4                  crayon_1.3.4                 
## [87] BiocFileCache_1.15.1          rmarkdown_2.6                
## [89] locfit_1.5-9.4                grid_4.1.0                   
## [91] data.table_1.13.4             blob_1.2.1                   
## [93] FNN_1.1.3                     HSMMSingleCell_1.11.0        
## [95] sparsesvd_0.2                 digest_0.6.27                
## [97] textshaping_0.2.1             munsell_0.5.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kelly Street, Koen Van den Berge, Hector Roux de Bezieux.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
